linker/shared/src/main/scala/org/scalajs/linker/frontend/optimizer/OptimizerCore.scala

```diff
@@ -2887,18 +2930,8 @@ private[optimizer] abstract class OptimizerCore(config: CommonPhaseConfig) {
       case Int_+ => IntLiteral(int(lhs) + int(rhs))
       case Int_- => IntLiteral(int(lhs) - int(rhs))
       case Int_* => IntLiteral(int(lhs) * int(rhs))
-
-      case Int_/ =>
-        int(rhs) match {
-          case 0 => IntLiteral(0) // Undefined Behavior
-          case r => IntLiteral(int(lhs) / r)
-        }
-
-      case Int_% =>
-        int(rhs) match {
-          case 0 => IntLiteral(0) // Undefined Behavior
-          case r => IntLiteral(int(lhs) % r)
-        }
+      case Int_/ => IntLiteral(int(lhs) / checkForIntDiv(int(rhs)))
+      case Int_% => IntLiteral(int(lhs) % checkForIntDiv(int(rhs)))
```

Comment:
Why do we need `checkForIntDiv`? Isn't this the very commit that make sure dividing by zero would throw?

Reply:
Yes, and that's why a division by 0 cannot be *constant-folded* anymore. It needs to preserved as a non-constant, which is why `constantFoldBinaryOp_except_String_+` cannot be called on a division by 0. Instead, that case is caught by `foldBinaryOp`.

Reply:
I understand that. But checkForIntDiv is simply an assertion so that if something tries to constant fold div by 0 it crashes (calling code guards against this). But there is no need for this, since the division itself will crash.

Reply:
True. We could simply let the division crash on its own. I don't have any preference. Do you prefer letting the division crash?

Reply:
Yeah, mostly because it is less code.

Reply:
Makes sense. Updated.